### PR TITLE
MUMMNG-1386: Messaging for widgets

### PR DESCRIPTION
**Pull request by jhanstra**
image

I'm not sure how this works across environments. Like, where is the new-stuff.json coming from in production? I may have implemented this wrong if there is a separate feed for each environment.

### CODE REVIEW

**Comment by vertein** [comment]
:+1: @jhanstra promised me that on screen size changes, it still works.

File: angularjs-portal-home/src/main/webapp/samples/new-stuff.json [hunk 1]
```diff
@@ -2,11 +2,11 @@
             [
               { 
                 "id" : 1,
-                "expireYr"  : 2016,
-                "expireMon" : 2,
-                "expireDay" : 20,
-                "title" : "AngularJS Portal",
-                "shortDesc" : "Welcome to the alternative home page to uPortal.",
+                "expireYr"  : 2015,
+                "expireMon" : 5,
```
Comment (timlevett) [inline angularjs-portal-home/src/main/webapp/samples/new-stuff.json]: Month is 0 based, fwiw.

http://www.w3schools.com/jsref/jsref_obj_date.asp

**Comment by timlevett** [comment]
New stuff is in the overlay for some environments. So just flag this issue with label mum-overlay.

I'm not 100% about the second desc hard coding. If we do accept this please create an issue to refactor/remove it later.

overall, :+1:

File: angularjs-portal-home/src/main/webapp/samples/new-stuff.json (hunk 1 above)
Comment (jhanstra) [inline angularjs-portal-home/src/main/webapp/samples/new-stuff.json]: Ah yes, good catch.

Comment (apetro) [inline angularjs-portal-home/src/main/webapp/samples/new-stuff.json]: Is there some reasonable way to comment this trip hazard?

File: angularjs-portal-frame/src/main/webapp/css/buckyless/home.less [hunk 2]
```diff
@@ -400,6 +400,18 @@ div.table-cell {
   width:100%;
   max-width:100%;
 }
+.header-opaque .second-description {
```
Comment (apetro) [inline angularjs-portal-frame/src/main/webapp/css/buckyless/home.less]: Is this "second-description", or is this maybe "smart-widget-selector-message" ?  Seems like semantically this is less about being a second description and more about being the specific magic incantation required to get that pointer to float pointing at that favorite mode selector control.

**Comment by apetro** [comment]
:+1:

**Comment by jhanstra** [comment]
@timlevett I removed the hard-coded code so we don't need to remember to take it out. It's kind of janky as it adds one-off attributes to the json feed but I don't think that's a huge deal. Let me know what you think.

**Comment by vertein** [comment]
I think I like it better in the JSON.  What happens when that second description isn't there?

**Comment by jhanstra** [comment]
@vertein it should just not show anything.

**Comment by vertein** [comment]
Cool, just checking. :)

**Comment by timlevett** [comment]
:+1:

**Pull request merged**